addons/im_livechat/models/im_livechat_channel_tag.py

```diff
@@ -0,0 +1,29 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, api, Command
+
+
+class Im_LivechatChannelTag(models.Model):
+    """Tags for Live Chat users."""
+
+    _name = "im_livechat.channel.tag"
+    _description = "Live Chat Channel Tags"
+    _order = "name"
+
+    name = fields.Char("Name", required=True)
+    color = fields.Integer(string="Color label")
+    discuss_channel_ids = fields.Many2many(
+        "discuss.channel",
+        "discuss_channel_livechat_channel_tag_rel",
+        ondelete='cascade',
+        string="Discuss Channels",
+    )
+
+    _name_unique = models.UniqueIndex("(name)")
+
+    @api.ondelete(at_uninstall=False)
+    def _unlink_sync_discuss_channel(self):
+        for tag in self:
+            tag.discuss_channel_ids.write({
+                "livechat_channel_tag_ids": [Command.unlink(tag.id)],
+            })
```

Comment:
Do we really need to do this manually? Aren't relation/fields automatically removed on delete?

Reply:
`_sync_field_names` is not handling this well as far as I tested

Reply:
I don't feel like it's a "fake" one, but remove relation before being deleted, which is making sense even though it would be better if we could handle this better if we make `_sync_field_names ` aware of this.

And by triggering `_sync_field_names` the bus can handle it along with other stuff and we don't have to have it with a custom bus.